### ISSUE TITLE
Feed add/edit screen does not show drop-down Creator Organisation if orgc_id = 0

### ISSUE THREAD

**Issue by chrisinmtown** [MISP/MISP#6979]
Feed add and edit screens do not display the drop-down Creator organisation if orgc_id = 0.

The Feed index page shows a column "Org".  If orgc_id is zero the feeds index page has no value for Org, just blank.  If orgc_id is non-zero then the index page shows the name corresponding to the orgc_id value.  All good so far.

So I click on the pencil in the action column and get to the Edit MISP Feed screen. The edit screen displays and allows selection using a drop-down box "Creator organisation" but ONLY if the orgc_value is *not zero*.

The add-feed screen similarly does not allow me, a system admin, to pick a Creator organisation. So after adding a feed by hand, the org column remains blank and I cannot select one.

And for what it's worth, using the UI I cannot remove the org entirely from a feed -- there is no "None" in the drop-down selection box.

If I have completely misunderstood how organizations work on Feeds, or there's something wrong with how my admin-privs user ID is set up, please educate me.  Thanks for listening.

### Work environment

| Questions                 | Answers
|---------------------------|--------------------
| Type of issue             | Bug
| OS version (server)       |  RedHat
| OS version (client)       | N/A
| PHP version               | 7.4
| MISP version / git hash   | 2.4.137
| Browser                   | N/A

### Expected behavior

The Add Feed and Edit Feed screens always display the Creator Organization drop-down.

### Actual behavior

The Edit Feed screen only displays the Creator Organization drop-down if a value other than None (orgc_id=0) is stored.  The Add Feed screen never displays that drop-down.

### Steps to reproduce the behavior

Click Feeds, then Add Feed.  Or Click Feeds, then edit a feed with no org shown.

**Comment by chrisinmtown** [comment]
For starters, would a CIRCL/MISP expert please explain how feed data item orgc_id is used?   Does MISP/Cake apply the orgc_id to all data items fetched & ingested from that feed?  If the orgc_id is not used, then this issue I've raised here is silly, but I sorta think it's being used SOMEWHERE.

Is this another aspect of feeds that's only documented in the admin training material?  I didn't see it among the 321 slides, could not find org_id either, but maybe I'm blind today.

**Comment by mokaddem** [comment]
Hey @chrisinmtown,
I haven't managed to reproduce the bug you experience on the latest version. For me, the `orgc` field can be picked normally.
Just to recap how this field is working:

`Orgc` can be specified only for the _CSV_ and _Freetext_ feed formats, and can be used if you want that another organisation than the owner takes the role of the creator.
For example, you can have the `Phishtank` feed having its `org` set to `CIRCL` but its `orgc` set to `Phishtank`.
You cannot set an `orgc` for feeds of the _MISP_ feed format as this field is intrinsically carried over along the feed.
I hope that helps.

**Comment by chrisinmtown** [comment]
@mokaddem thanks for explaining that orgc can only be set for CSV/Freetext formats.  That means I misused the feed screen. I tried adding a new feed again and yep, if I set the format to one of those, the Organisation drop-down appears and I can pick a value. Same goes for all existing feeds with those formats. Sorry to waste your time.

p.s. please document this behavior in misp-book.

**Comment by chrisinmtown** [comment]
@mokaddem I read your explanation a few more times but still don't understand.  Your statement makes me think there's an "ORG" and "ORGC" for feeds. But on the UI feeds screen I only see a single column labeled "Org" and it is populated from the orgc_id field behind the scenes. On the UI an entry in the organisation field is absolutely required for feed types csv/freetext. Also I checked the table and it has only orgc_id column.  Am I missing something blindingly obvious?  Please explain more.

**Comment by mokaddem** [comment]
Whenever a feed (whichever format) is pulled in MISP, its data will create on (or more) event(s).
Depending on the feed format, you have two cases:
1. **MISP** feed format
    - When the events are created, the `org_id` (owner) of the event will be the `ID` for the organisation triggering the pull
    - the `orgc_id` will be the `ID` of the creator organisation defined in the feed itself. This is why it cannot be overridden by the feed configuration as it is already embedded in the data itself
2. **CSV**/**Freetext** feed format
    - In contrary to the **MISP** feed format, these do not ship the creator organisation along.
    - As every event in MISP must have a creator organisation `orgc_id`, you can set it in the feed configuration

TL;DR, `orgc` in the feed configuration can set (or override) the orgc_id when events are created from the feed's data
Does it make sense?

**Comment by chrisinmtown** [comment]
@mokaddem thanks for the additional explanation.  One more question, if you can please find time. We trigger fetches of CSV/freetext feeds using crontab, which invokes Cake with a  feed id *and* a user ID.  Every one of those feeds has an orgc_id in the feeds table, selected via the Feeds UI screen. How does Cake use the user ID and the orgc_id when it processes feeds -- where/how are those IDs applied to events and attributes?

**Comment by mokaddem** [comment]
When a CSV/freetext feed is pulled, an event is created and saved.
These fields are set as follow:
- `orgc_id`: Taken from the feed configuration `orgc_id` field.
- `org_id`: Taken from the user performing the pull request. So in your case, the `user_id` is used to fetch his organisation (and thus the associated `org_id`).

**Comment by chrisinmtown** [comment]
I tried to capture the answer from @mokaddem in https://github.com/MISP/misp-book/pull/225 please review
